app/src/main/java/ai/elimu/analytics/util/VersionHelper.kt

```diff
@@ -146,4 +146,35 @@ object VersionHelper {
             SharedPreferencesHelper.storeAppVersionCode(context, newVersionCode)
         }
     }
+
+    /**
+     * Extracts the version code as an Int from a string with pattern:
+     * <prefix>_<versionCode>_<description>_<yyyy-MM-dd>.csv
+     * Example: a1b2c3d4e5f6g7h8_4004008_word-learning-events_2025-06-01.csv -> 4004008
+     *
+     * @return The version code as an Int, or null if the string is invalid.
+     */
+    fun String.extractVersionCode(): Int? {
```

Comment:
@tuancoltech By the way, this same function is also used in the backend for processing the data uploaded via the backend's REST API. https://github.com/elimu-ai/webapp/blob/main/src/main/java/ai/elimu/util/AnalyticsHelper.java

So we might want to move this code into a shared library that both the Analytics app and the Webapp can use.

After this PR merges, we can move this code (and the code in the `webapp` repo) in to the shared `model` library: 
* https://github.com/elimu-ai/model/issues/356